openmls/src/framing/test_framing.rs

```diff
@@ -448,7 +448,7 @@ fn unknown_sender<Provider: OpenMlsProvider>(ciphersuite: Ciphersuite, provider:
 
     let params = CreateCommitParams::builder()
         .framing_parameters(framing_parameters)
-        .proposal_store(&proposal_store)
+        .proposal_store(group_alice.proposal_store())
```

Comment:
At the risk of creating more churn: Do we still need to pass the proposal store in the CreateCommitParams? Can't create_commit() just use the internal proposal store directly?

Doesn't have to be fixed with this PR, though.

Reply:
The current tests should already require that all parties have separate providers, so I agree, there shouldn't be a need to pass it in.

Reply:
I did the proposed change and it worked as expected. I had to do all the test refactoring that I tried to avoid in the first place though. This made me also discover a bug in the public group: merging a commit didn't empty the proposal store, this is now fixed.